### PR TITLE
start at selection tutorial

### PR DESCRIPTION
**Pull request by petrelharp**
I've moved the code from `docs/selection_example.py` over into the Tutorials section. It needs a lot of improvements.  But since we want some people to start using this stuff, maybe we should merge it as-is and improve?

To see these docs, check out this PR; go to the `docs/` directory and run `make`; then point your web browser at `docs/_build/html`.

### CODE REVIEW

**Review by grahamgower — APPROVED** [review]
LGTM!

**Comment by petrelharp** [comment]
@mufernando is going to add something to this, and I'm hoping to take a pass through with some more explanation, so let's hold off on merging it for a minute.

**Comment by petrelharp** [comment]
The annotation part of this is not working: see #952.

**Comment by mufernando** [comment]
- [ ] remove annotation example
- [ ] add sweep example, conditioning of freq also

**Comment by mufernando** [comment]
I've been trying to add a simpler example with the DrawMutation/etc API and I can't figure this out. 

First, I noticed the example we had was not working. This was because we were forgetting to add the mutation type to the contig. So I added a new method to easily add mutation types not tied to a genomic element type.

But still, the example under "selective sweeps" in the tutorial is not working. No mutation is placed at that site.

@grahamgower

**Comment by grahamgower** [comment]
That's very strange. I'm fairly sure we have tests that check the frequency of the drawn allele after conditioning. Are these tests broken in some way that silently prevents them from running?

**Comment by grahamgower** [comment]
I think what's going on here is that the allele is present in the population (at low frequency), but not present in the 10 sampled individuals.

And assuming you sample an individual with the drawn mutation, the following code overrides the mutation types for your example code, so the mutation's `selection_coeff` metadata is set to zero.
https://github.com/popsim-consortium/stdpopsim/blob/81bc0144bb144c83cfcd2393bd235872dbc6a257/stdpopsim/slim_engine.py#L1143-L1145

**Comment by mufernando** [comment]
@petrelharp I added an example of a selective sweep. this looks good to me, but you said you wanted to do another pass.

One last thing we should showcase here is the ChangeMutationFitness, which can be used to simulate soft sweeps.

File: docs/tutorial.rst
Comment (petrelharp) [inline docs/tutorial.rst:1101]: This conditioning is redundant - it can't have been lost if it's at frequency above 80% at the end. Is the purpose to restart the simulation earlier, if it's lost?

Comment (mufernando) [reply docs/tutorial.rst:1101]: yeah, the purpose is so that we don't wait until the end to check, instead we will be checking for presence every generation.

File: docs/tutorial.rst
Comment (petrelharp) [inline docs/tutorial.rst:1093]: how does it know to restart from this point? Does it just go back to the last saved point?

**Comment by petrelharp** [comment]
Hm: I see the extended events are not yet documented.

And - I've split up the last example by adding more words. Maybe this is helpful?

File: docs/tutorial.rst
Comment (mufernando) [inline docs/tutorial.rst:1115]: it would be + 1 because this API has forward times

Comment (petrelharp) [reply docs/tutorial.rst:1115]: good catch - I meant to ask. If it has forwards times, though, why are "start time" and "end time" equal to 0 for the final conditioning?

Comment (grahamgower) [reply docs/tutorial.rst:1115]: The API here is confusing, sorry. Times have units of "generations ago", just like msprime. The start_time/end_time parameters for the `ChangeMutationFitness` and `ConditionOnAlleleFrequency` need `start_time => end_time`, which is opposite to what msprime uses (e.g in the demography debugger). I'm not sure why I did it that way.

There are various comments and docstring stubs in https://github.com/popsim-consortium/stdpopsim/blob/main/stdpopsim/ext/selection.py which explain some things.

Comment (petrelharp) [reply docs/tutorial.rst:1115]: Ah-ha, thanks. ... so, should it have been `T-1` or `T+1` there?

File: docs/tutorial.rst
Comment (grahamgower) [inline docs/tutorial.rst:1123]: This doesn't look right. I think this should be appending an element, not a list containing one element.

File: docs/tutorial.rst
Comment (grahamgower) [inline docs/tutorial.rst:1141]: Likewise here, I think it shouldn't be a list.

File: docs/tutorial.rst
Comment (grahamgower) [inline docs/tutorial.rst:1200]: Cool! It works!

File: docs/tutorial.rst
Comment (grahamgower) [reply docs/tutorial.rst:1115]: `T-1` looks right to me.

Comment (mufernando) [reply docs/tutorial.rst:1115]: gee, sorry for the confusion!

File: docs/tutorial.rst
Comment (mufernando) [reply docs/tutorial.rst:1200]: I wanted sphinx to render the plot using matplotlib, but it looks like the page that describes how to do this hasn't been updated in many years and the instructions don't work anymore.

File: docs/tutorial.rst
Comment (mufernando) [reply docs/tutorial.rst]: I think it goes back to the last saved point. @grahamgower

File: docs/tutorial.rst
Comment (petrelharp) [reply docs/tutorial.rst:1199]: if/when we switch to jupyterbook then this does work...

**Comment by petrelharp** [comment]
Thanks - fixed those things (and actually ran the code =). I'll let you all merge it if you think it's ready?

File: docs/tutorial.rst
Comment (grahamgower) [reply docs/tutorial.rst]: Yep, exactly.

**Comment by grahamgower** [comment]
> Thanks - fixed those things (and actually ran the code =). I'll let you all merge it if you think it's ready?

Looks good. Maybe it should be squashed though?

**Review by mufernando — APPROVED** [review]
LGTM!

**Pull request merged**
